### ISSUE TITLE
RegisterResolveInterceptor isn't hit for collections

### ISSUE THREAD

**Issue by dotnetjunkie** [simpleinjector/SimpleInjector#742]
**Describe the bug**

The following code doesn't print the name of the type to the console.

``` c#
var container = new Container();
            
container.Options.RegisterResolveInterceptor((c, p) =>
{
    System.Console.WriteLine(c.Registration.ImplementationType.ToFriendlyName());
    return p();
},
c => true);

container.Collection.Append<ITimeProvider, RealTimeProvider>();
            
container.GetInstance<IEnumerable<ITimeProvider>>();
```

**Expected behavior**

The name of the type should be written to the console.

**Actual behavior**

The following code doesn't print the name of the type to the console.

**Additional context**
Latest release v4.6.2.

**Comment by dotnetjunkie** [comment]
[feature-742](https://github.com/simpleinjector/SimpleInjector/tree/feature-742) branch created.

**Comment by dotnetjunkie** [comment]
Unable to reproduce.  The problem might already have been fixed somewhere after v4.6.2. Will add unit tests to make sure this keeps on working.
